File: storagechooser/src/main/java/com/codekidlabs/storagechooser/utils/ResourceUtil.java
package com.codekidlabs.storagechooser.utils;


import android.content.Context;
import androidx.core.content.ContextCompat;
import androidx.core.graphics.ColorUtils;

import com.codekidlabs.storagechooser.R;

public class ResourceUtil {

    private Context context;

    public ResourceUtil(Context context) {
        this.context = context;
    }

    public int getColor(int id) {
        return ContextCompat.getColor(context, id);
    }

    public int getAppliedAlpha(int color) {
        return ColorUtils.setAlphaComponent(color, 50);
    }

    public int getPrimaryColorWithAlpha() {
        return getAppliedAlpha(getColor(R.color.colorPrimary));
    }
}

File: storagechooser/src/main/java/com/codekidlabs/storagechooser/utils/ResourceUtil.kt
package com.codekidlabs.storagechooser.utils


import android.content.Context
import androidx.core.content.ContextCompat
import androidx.core.graphics.ColorUtils

import com.codekidlabs.storagechooser.R

class ResourceUtil(private val context: Context) {

    val primaryColorWithAlpha: Int
        get() = getAppliedAlpha(getColor(R.color.colorPrimary))

    fun getColor(id: Int): Int {
        return ContextCompat.getColor(context, id)
    }

    fun applyAlpha(color: Int, alpha: Int) = ColorUtils.setAlphaComponent(color, alpha)

    fun getAppliedAlpha(color: Int): Int {
        return ColorUtils.setAlphaComponent(color, 50)
    }
}
